groups/bsl/bslstl/bslstl_optional.t.cpp

```diff
@@ -758,120 +980,11218 @@ struct UsesAllocatorArgT<MyClass2a> : bsl::true_type {
+        // be executed with types that are constructible from an integer, have
+        // a 'value()' method, and for which there exists an overload of
+        // 'isConstRef'.
+
+    static void testCase11();
+        // TESTING 'operator->' MEMBER FUNCTION.  Note that this test can only
+        // be executed with types that are constructible from an integer and
+        // have a 'value()' method.
+
+    static void testCase10();
+        // TESTING ALLOCATOR EXTENDED 'value_or' METHOD.  Note that this test
+        // can only be executed with types that are constructible from an
+        // integer, have a 'value()' method, are allocator-aware and have a
+        // 'get_allocator' method.
+
+    static void testCase9();
+        // TESTING 'value_or' METHOD.  Note that this test can only be executed
+        // with types that are constructible from an integer, have a 'value()'
+        // method, and, if TYPE is allocator-aware, has a 'get_allocator'
+        // method
+
+    static void testCase8();
+        // TESTING 'value' METHOD.  Note that this test can only be executed
+        // with types that are constructible from an integer, have a 'value()'
+        // method, and for which there exists an overload of 'isRvalueRef' and
+        // 'isConstRef'.
+
+    static void testCase7_imp();
+    static void testCase7();
+        // TESTING 'reset' MEMBER FUNCTION.  Note that this test requires
+        // <TYPE> to provide a 's_destructorInvocations' static variable which
+        // counts the number of times destructor has been invoked.
+
+    static void testCase6();
+        // TESTING CONVERSION TO BOOL
+
+    static void testCase5();
+        // ALLOCATOR EXTENDED DISENGAGED CONSTRUCTORS.  Note that this test
+        // requires 'TYPE' to be allocator-aware.
+
+    static void testCase4();
+        // DISENGAGED CONSTRUCTORS AND DESTRUCTOR.  Note that this test
+        // requires <TYPE> to provide a 's_destructorInvocations' static
+        // variable which counts the number of times destructor has been
+        //invoked.
+
+    static void testCase3();
+        // BREATHING TEST
+
+    static void testCase2();
+        // Optional_DataImp TEST.  The test requires TYPE to be default
+        // constructible.
+
+    static void testCase1();
+        // Optional_Data TEST
+};
+
+template <class TYPE>
+void bslstl_optional_value_type_deduce(const bsl::optional<TYPE>&)
+{
+}
+
+template <class TYPE>
+void bslstl_optional_optional_type_deduce(const TYPE&)
+{
+}
+
+template <class TYPE>
+void TestDriver<TYPE>::testCase33()
+{
+    // --------------------------------------------------------------------
+    // TESTING ALLOCATOR EXTENDED 'make_optional' USE CASES
+    //
+    //  In this test, we check that there are no additional copies of the
+    //  newly created 'optional' object when using the allocator extended
+    //  'make_optional' even before RVO was mandatory.
```

Comment:
```suggestion
    //  Ensure that there are no additional copies of the
    //  newly created 'optional' object when using the allocator extended
    //  'make_optional' even before RVO was mandatory.
```

Same in other places.

Reply:
done